pkg/event/controller.go

```diff
@@ -95,9 +113,30 @@ func (gen *generator) Run(ctx context.Context, workers int, waitGroup *sync.Wait
 }
 
 func (gen *generator) startRecorders(ctx context.Context) error {
-	if err := gen.broadcaster.StartRecordingToSinkWithContext(ctx); err != nil {
+	eventHandler := func(obj runtime.Object) {
+		event, ok := obj.(*apieventsv1.Event)
+		if !ok {
+			gen.logger.Error(nil, "unexpected type, expected eventsv1.Event")
+			return
+		}
+
+		eventCopy := event.DeepCopy()
+		eventCopy.ResourceVersion = ""
+		_, err := gen.eventsClient.Events(event.Namespace).Create(ctx, eventCopy, metav1.CreateOptions{})
+		if err != nil {
+			gen.droppedEventsCounter.Add(ctx, 1)
+			gen.logger.Error(err, "failed to create event")
+		}
+	}
+	stopWatcher, err := gen.broadcaster.StartEventWatcher(eventHandler)
```

Comment:
This bypasses all the broadcaster logic.

Reply:
@eddycharly Yes, but that logic is only the aggregating isomorphic events with the same key to form an EventSeries. It's supposed to help with performance but in our case the performance is worse. We lose EventSeries only if two events near each other in time have the same:

1. eventType
2. action
3. reason
4. reportingController
5. reportingInstance
6. regarding